src/dsa/rpo_falcon512/math/samplerz.rs

```diff
@@ -0,0 +1,298 @@
+use core::f64::consts::LN_2;
+use rand::Rng;
+
+#[cfg(not(feature = "std"))]
+use num::Float;
```

Comment:
Once I got rid of the `getrandom` feature/dependency, I had to add `libm` feature for the `num` crate and then insert these lines in several files. I didn't look deeply enough to figure out if there is a better way to do it (I don't mind using `libm` feature on `num` but adding these lines is a little annoying). Maybe @bitwalker would have a suggestion here?

Reply:
Also, maybe good to enable `libm` feature for `num` only for `no_std` builds - but I wasn't sure how to do this.

Reply:
Is it actually necessary for it to be conditionally imported vs just always importing it? My usual recommendation would be to just always import things like that, and only make it conditional if the compiler complains about it (either due to being unused, in which case you _could_ just disable the warning on that import, but that's not much different than conditionally importing it; or due to conflict/error in certain feature configurations).

I would think in the case of `num::Float`, being a trait, it would be safe to always import it - but I suppose the compiler might complain if it's only needed by some _other_ conditionally-compiled code. In which case, it probably makes more sense to move this import into that scope instead. For example, if you have a function that is conditionally enabled, and that is the function that requires this import, you can just move the `use num::Float;` into the body of the function, which is easier to maintain anyway for things like this.

In some cases, having a `prelude` module that you can just wildcard import from throughout the crate can solve papercuts like this, but the downside of that is a loss of clarity as to what is in scope in a given module. Basically tradeoffs either way.

Reply:
Yeah - I tried to remove the conditional flag but got warnings. So, I left them as is.